usr/src/test/zfs-tests/tests/functional/trim/trim.kshlib

```diff
@@ -0,0 +1,110 @@
+#!/bin/ksh -p
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright (c) 2017 by Tim Chase. All rights reserved.
+# Copyright (c) 2017 by Nexenta Systems, Inc. All rights reserved.
+#
+
+# Adaptation from older zfs-tests
+if [ -z "$ZPOOL" ]; then
+	ZPOOL=zpool
+fi
+if [ -z "$ZFS" ]; then
+	ZFS=zfs
+fi
+if [ -z "$FILE_WRITE" ]; then
+	FILE_WRITE=file_write
+fi
+
+function set_tunable64
+{
+	set_tunable_impl "$1" "$2" Z
+}
+
+function set_tunable32
+{
+	set_tunable_impl "$1" "$2" W
+}
+
+function set_tunable_impl
+{
+	typeset tunable="$1"
+	typeset value="$2"
+	typeset mdb_cmd="$3"
+
+	[[ -z "$tunable" ]] && return 1
+	[[ -z "$value" ]] && return 1
+	[[ -z "$mdb_cmd" ]] && return 1
+
+	case "$(uname)" in
+	Linux)
+		typeset zfs_tunables="/sys/module/zfs/parameters"
+		[[ -f "$zfs_tunables/$tunable" ]] || return 1
+		echo -n "$value" > "$zfs_tunables/$tunable"
+		return "$?"
+		;;
+	SunOS)
+		echo "${tunable}/${mdb_cmd}0t${value}" | mdb -kw
+		return "$?"
+		;;
+	esac
+}
+
+function setupvdevs
+{
+	log_must rm -f $VDEVS
+	log_must truncate -s $VDEV_SIZE $VDEVS
+}
+
+function getsizemb
+{
+	case "$(uname)" in
+	Linux)
+		typeset rval
+		rval=$(du --block-size 1048576 -s "$1" | sed -e 's;[ <->].*;;')
```

Comment:
I'm not sure what the sed is trying to do here.  The du command appears to separate the size from the name by a single tab so rather than embedding a tab in the sed, I just used awk as is the case below.